src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -487,6 +499,10 @@ static inline JSC::EncodedJSValue jsBufferConstructorFunction_allocBody(JSC::JSG
 
     double lengthDouble = lengthValue.toIntegerWithTruncation(lexicalGlobalObject);
 
+    if (UNLIKELY(lengthDouble != lengthDouble)) {
+        throwNodeRangeError(lexicalGlobalObject, scope, "Buffer size must be 0...9007199254740991, got NaN"_s);
```

Comment:
can just make throwRangeError do this instead of adding another

Reply:
this is calling the same method with a different message, not a new function